extensions/amp-list/0.1/amp-list.js

```diff
@@ -386,16 +389,71 @@ export class AmpList extends AMP.BaseElement {
       const event = createCustomEvent(this.win,
           AmpEvents.DOM_UPDATE, /* detail */ null, {bubbles: true});
       this.container_.dispatchEvent(event);
-
       // Change height if needed.
       this.measureElement(() => {
         const scrollHeight = this.container_./*OK*/scrollHeight;
         const height = this.element./*OK*/offsetHeight;
         if (scrollHeight > height) {
-          this.attemptChangeHeight(scrollHeight).catch(() => {});
+          if (autoResize) {
+            // This needs to be gated behind the experiment 'amp-list-resize'
+            const layout = this.element.getAttribute('layout');
+            if (layout == Layout.NODISPLAY || layout == Layout.FILL) {
+              return;
+            } else if (layout == Layout.FLEX_ITEM) {
+              // TODO (cathyxz): flex item does not play nice
+              // with layout container
+              this.attemptChangeHeight(scrollHeight).catch(() => {});
+            } else if (layout !== Layout.CONTAINER) {
+              this.changeToLayoutContainer_();
+            }
+          } else {
+            this.attemptChangeHeight(scrollHeight).catch(() => {});
+          }
         }
       });
-    }, this.container_);
+    });
+  }
+
+  /**
+   * Converts the amp-list to de facto layout container.
+   * @private
+   */
+  changeToLayoutContainer_() {
+    const layout = this.element.getAttribute('layout');
+    if (layout == Layout.RESPONSIVE) {
+      this.element.classList.remove('i-amphtml-layout-responsive');
+    } else if (layout == Layout.FIXED) {
+      this.element.classList.remove('i-amphtml-layout-fixed');
+      setStyles(this.element, {
+        height: '',
+      });
+    } else if (layout == Layout.FIXED_HEIGHT) {
+      this.element.classList.remove('i-amphtml-layout-fixed-height');
+      setStyles(this.element, {
+        height: '',
+        width: '',
+      });
+    } else if (layout == Layout.INTRINSIC) {
+      this.element.classList.remove('i-amphtml-layout-intrinsic');
+    }
+    const sizer = childElementByTag(this.element, 'i-amphtml-sizer');
+    if (sizer) {
+      this.element.removeChild(sizer);
+    }
+    this.element.classList.remove('i-amphtml-layout-size-defined');
+    this.container_.classList.remove(
+        'i-amphtml-fill-content',
+        'i-amphtml-replaced-content'
```

Comment:
I'm not sure we want to remove these (undoing `applyFillContent`). Do we need to?

Reply:
Yes we do if we are setting the `<amp-list>` to layout container. `i-amphtml-fill-content` basically does this: 
```
.i-amphtml-fill-content {
    display: block;
    height: 0;
    max-height: 100%;
    max-width: 100%;
    min-height: 100%;
    min-width: 100%;
    width: 0;
    margin: auto;
}
```

Since `<amp-list>` is now layout container, it results in the `<amp-list>` having a height/width of 0, and the contents of `<amp-list>` become overflow elements.